### PR TITLE
Add wait for it script to ensure db ready before starting backend

### PR DESCRIPTION
**Pull request by Jorgro**
## Proposed Changes
- Add [wait-for-it.sh](https://github.com/vishnubob/wait-for-it) to `entrypoint.sh` to ensure that the DB is available before the Backend tries to connect. While there is a `depends_on` in the Docker Compose file this does **not** ensure that the DB is ready, only that the DB container is alive. I experienced several times that the DB would start after the Backend and the Backend would never try to reconnect, this will atleast stop that from happening. 

## Types of Changes

- [ ] New feature
- [x] Bug fix
- [ ] Enhancement/optimization
- [ ] Refactor
- [ ] Style
- [ ] Build/dependencies
- [ ] Other

## Checklist

- [x] I have added tests to cover my changes (for features/bugs)
- [x] I am pleased with the readability of the code (if not, state where you need input)
- [x] I have tested the changes and verified that they work and do not break anything (to the best of my ability)

### CODE REVIEW

**Comment by Jorgro** [comment]
It is also possible to install the `wait-for-it` in the docker container as it is an debian package: https://tracker.debian.org/pkg/wait-for-it. Do we prefer that or adding the script manually like I did here?

**Comment by Jorgro** [comment]
An alternative solution would be something like this: https://stackoverflow.com/questions/31746182/docker-compose-wait-for-container-x-before-starting-y

**Comment by larwaa** [comment]
```dockerfile
depends_on:
    condition:
```
Is no longer document as of [Docker Compose v3](https://docs.docker.com/compose/compose-file/compose-file-v3/#depends_on). As this is only used for local development, one could alternatively add a short sleep script to our existing `entrypoint.sh` to deal with this issue. However, why bother when we can utilize `wait-for-it.sh`?

File: backend/wait-for-it.sh [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+if [[ $WAITFORIT_CLI != "" ]]; then
+    if [[ $WAITFORIT_RESULT -ne 0 && $WAITFORIT_STRICT -eq 1 ]]; then
+        echoerr "$WAITFORIT_cmdname: strict mode, refusing to execute subprocess"
+        exit $WAITFORIT_RESULT
+    fi
+    exec "${WAITFORIT_CLI[@]}"
+else
+    exit $WAITFORIT_RESULT
+fi
```
Comment (larwaa) [inline backend/wait-for-it.sh:183]: newline

**Review by larwaa — CHANGES_REQUESTED** [review]

File: backend/wait-for-it.sh [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+if [[ $WAITFORIT_CLI != "" ]]; then
+    if [[ $WAITFORIT_RESULT -ne 0 && $WAITFORIT_STRICT -eq 1 ]]; then
+        echoerr "$WAITFORIT_cmdname: strict mode, refusing to execute subprocess"
+        exit $WAITFORIT_RESULT
+    fi
+    exec "${WAITFORIT_CLI[@]}"
+else
+    exit $WAITFORIT_RESULT
```
Comment (Jorgro) [inline backend/wait-for-it.sh:182]:
```suggestion
    exit $WAITFORIT_RESULT
    
```

**Review by Jorgro — COMMENTED** [review]
good work! keep it up

**Pull request merged**